storage/sequence_cache.go

```diff
@@ -117,55 +141,53 @@ func (rc *SequenceCache) CopyInto(e engine.Engine, destRangeID roachpb.RangeID)
 // locked while copying is in progress. Failures decoding individual
 // entries return an error. The copy is done directly using the engine
 // instead of interpreting values through MVCC for efficiency.
-func (rc *SequenceCache) CopyFrom(e engine.Engine, originRangeID roachpb.RangeID) error {
-	start := engine.MVCCEncodeKey(
-		keys.SequenceCacheKey(originRangeID, roachpb.KeyMin))
-	end := engine.MVCCEncodeKey(
-		keys.SequenceCacheKey(originRangeID, roachpb.KeyMax))
-
-	return e.Iterate(start, end, func(kv engine.MVCCKeyValue) (bool, error) {
-		// Decode the key into a cmd, skipping on error. Otherwise,
-		// write it to the corresponding key in the new cache.
-		family, err := rc.decodeSequenceCacheKey(kv.Key)
-		if err != nil {
-			return false, util.Errorf("could not decode a sequence cache key %s: %s",
-				roachpb.Key(kv.Key), err)
-		}
-		key := keys.SequenceCacheKey(rc.rangeID, family)
-		encKey := engine.MVCCEncodeKey(key)
-		// Decode the value, update the checksum and re-encode.
-		meta := &engine.MVCCMetadata{}
-		if err := proto.Unmarshal(kv.Value, meta); err != nil {
-			return false, util.Errorf("could not decode sequence cache value %s [% x]: %s",
-				roachpb.Key(kv.Key), kv.Value, err)
-		}
-		meta.Value.Checksum = nil
-		meta.Value.InitChecksum(key)
-		_, _, err = engine.PutProto(e, encKey, meta)
-		return false, err
-	})
+func (sc *SequenceCache) CopyFrom(e engine.Engine, originRangeID roachpb.RangeID) error {
```

Comment:
`CopyFrom` and `CopyInto` look nearly identical; why do we need both? Can't they at least share the Iterate closure?